purchases/src/main/kotlin/com/revenuecat/purchases/identity/IdentityManager.kt

```diff
@@ -49,10 +55,16 @@ internal class IdentityManager(
             ?: deviceCache.getLegacyCachedAppUserID()
             ?: generateRandomID()
         log(LogIntent.USER, IdentityStrings.IDENTIFYING_APP_USER_ID.format(appUserIDToUse))
-        deviceCache.cacheAppUserID(appUserIDToUse)
-        subscriberAttributesCache.cleanUpSubscriberAttributeCache(appUserIDToUse)
-        deviceCache.cleanupOldAttributionData()
-        invalidateCustomerInfoAndETagCacheIfNeeded(appUserIDToUse)
+
+        val cacheEditor = deviceCache.startEditing()
+        deviceCache.cacheAppUserID(appUserIDToUse, cacheEditor)
+        subscriberAttributesCache.cleanUpSubscriberAttributeCache(appUserIDToUse, cacheEditor)
+        invalidateCustomerInfoAndETagCacheIfNeeded(appUserIDToUse, cacheEditor)
+        cacheEditor.apply()
+
+        enqueue {
+            deviceCache.cleanupOldAttributionData()
```

Comment:
Hmm doing this on a thread could be dangerous I think? As in, this will remove all attribution, so there is a small chance of actual new data making it in before we delete it, then deleting it

Reply:
note this is attribution data and not subscriber attributes. Not sure if you were at RevenueCat when we replaced  attribution data with subscriber attributes. Attribution data is deprecated and we don't actually ever post anything in the key. I thought we were still using this for the device identifiers, but we are using subscriber attributes for that.

This function is just cleaning up the caches in case someone is upgrading from an old version of the SDK that used attribution data

Reply:
Ohhh that makes total sense then! Thanks for explaining!